src/tools/crossgen2/ILCompiler.ReadyToRun/Compiler/DependencyAnalysis/ReadyToRunSymbolNodeFactory.cs

```diff
@@ -271,116 +394,126 @@ private ISymbolNode CreateMethodDictionary(MethodWithToken method, SignatureCont
                     signatureContext));
         }
 
-        private readonly Dictionary<FieldDesc, ISymbolNode> _fieldAddressCache = new Dictionary<FieldDesc, ISymbolNode>();
-
-        public ISymbolNode FieldAddress(FieldDesc fieldDesc, SignatureContext signatureContext)
+        private struct FieldAndSignatureContext : IEquatable<FieldAndSignatureContext>
         {
-            ISymbolNode result;
-            if (!_fieldAddressCache.TryGetValue(fieldDesc, out result))
+            public readonly FieldDesc Field;
+            public readonly SignatureContext SignatureContext;
+
```

Comment:
I think now that we use .Equals and .GetHashCode on SignatureContext objects, we should probably make it an IEquatable class and provide our implementation of these 2 APIs.

Reply:
Yes, seems like SignatureContext is not interned and we're potentially making lots of new ones. We should override Equals and GetHashCode.

I tend to only implement IEquatable if it's a struct (to avoid boxing). It doesn't make much difference for classes.